Chaincase.UI/ViewModels/ConnectNodeViewModel.cs

```diff
@@ -0,0 +1,136 @@
+using System.Net;
+using WalletWasabi.Exceptions;
+using System.Globalization;
+using System.Threading;
+using NBitcoin.Protocol;
+using NBitcoin.Protocol.Behaviors;
+
+namespace Chaincase.UI.ViewModels
+{
+    public class ConnectNodeViewModel : ReactiveObject
+    {
+        private readonly Global _global;
+        private readonly ChaincaseWalletManager _walletManager;
+        private readonly Config _config;
+        private readonly UiConfig _uiConfig;
+        private readonly FeeProviders _feeProviders;
+        private readonly ChaincaseSynchronizer _synchronizer;
+
+        private string _nodeAddress;
+        private bool _hasSetCustomAddress;
+        private bool? _canConnected;
+
+        public ConnectNodeViewModel(Global global, ChaincaseWalletManager walletManager, Config config, ChaincaseSynchronizer synchronizer)
+        {
+            _global = global;
+            _walletManager = walletManager;
+            _config = config;
+            _synchronizer = synchronizer;
+            _hasSetCustomAddress = _config.GetBitcoinP2pEndPoint().ToString() != "127.0.0.1:8333";
+            if (_hasSetCustomAddress) {
+                _nodeAddress = _config.GetBitcoinP2pEndPoint().ToString();
+                try
+                {
+                    SetNodeAddress();
+                }
+                catch (Exception _) {
+                    _canConnected = false;
+                }
+            }
+        }
+
+        public void SetDefaultNodeAddress() {
+            if (_config.Network == NBitcoin.Network.Main)
+            {
+                _config.SetP2PEndpoint(CreateEndPoint(FormatHostAndPort(_config.DefaultMainNetP2PHost, _config.DefaultMainNetP2PPort)));
+            }
+            else if (_config.Network == NBitcoin.Network.TestNet)
+            {
+                _config.SetP2PEndpoint(CreateEndPoint(FormatHostAndPort(_config.DefaultTestNetP2PHost, _config.DefaultTestNetP2PPort)));
+            }
+            else {
+                throw new NotSupportedNetworkException(_config.Network);
+            }
+
+            _hasSetCustomAddress = false;
+            _canConnected = false;
+        }
+
+        public void SetNodeAddress() {
+            // Timeout after 10 seconds
+            var cancellationToken = new CancellationToken();
+            using var handshakeTimeout = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+            handshakeTimeout.CancelAfter(TimeSpan.FromSeconds(10));
+            var nodeConnectionParameters = new NodeConnectionParameters()
+            {
+                ConnectCancellation = handshakeTimeout.Token,
+                IsRelay = false,
+            };
+            nodeConnectionParameters.TemplateBehaviors.Add(new SocksSettingsBehavior(_synchronizer.WasabiClient.TorClient.TorSocks5EndPoint, onlyForOnionHosts: true, networkCredential: null, streamIsolation: false));
+
+            Node.Connect(_config.Network, _nodeAddress, nodeConnectionParameters);
```

Comment:
Don't you also need to add the node to `P2pBlockProvider`  somehow? The ctor takes a `CoreNode` param, to which it manages to extract  a `Node` for  `Node _localBitcoinCoreNode`